lms/envs/common.py

```diff
@@ -2295,3 +2295,6 @@
 PROFILE_IMAGE_SECRET_KEY = 'placeholder secret key'
 PROFILE_IMAGE_MAX_BYTES = 1024 * 1024
 PROFILE_IMAGE_MIN_BYTES = 100
+
+# This is to check the domain in case of preview.
+PREVIEW_DOMAIN = 'preview'
```

Comment:
@smaeda-ks We are simplifying this logic in https://github.com/edx/edx-platform/pull/11661 so that only the `PREVIEW_LMS_BASE` needs to be set.

I agree that the preview functionality should only be accessible to course staff. I have created a ticket to track that: https://openedx.atlassian.net/browse/TNL-4194.